bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/discovery/RemoteopenhabDiscoveryParticipant.java

```diff
@@ -0,0 +1,89 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.remoteopenhab.internal.discovery;
+
+import static org.openhab.binding.remoteopenhab.internal.RemoteopenhabBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.jmdns.ServiceInfo;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.remoteopenhab.internal.config.RemoteopenhabInstanceConfiguration;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RemoteopenhabDiscoveryParticipant} is responsible for discovering
+ * the remote openHAB servers using mDNS discovery service.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = MDNSDiscoveryParticipant.class, configurationPid = "mdnsdiscovery.remoteopenhab")
+public class RemoteopenhabDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteopenhabDiscoveryParticipant.class);
+
+    private static final String SERVICE_TYPE = "_openhab-server._tcp.local.";
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(ServiceInfo service) {
+        // We use the first host address as thing ID
+        if (getServiceType().equals(service.getType()) && service.getHostAddresses() != null
+                && service.getHostAddresses().length > 0 && !service.getHostAddresses()[0].isEmpty()) {
+            return new ThingUID(BRIDGE_TYPE_SERVER, service.getHostAddresses()[0].replaceAll("[^A-Za-z0-9_]", "_"));
+        }
+        return null;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        logger.debug("createResult ServiceInfo: {}", service);
+        DiscoveryResult result = null;
+        String url = null;
+        if (service.getURLs() != null && service.getURLs().length > 0 && !service.getURLs()[0].isEmpty()) {
+            url = service.getURLs()[0];
+        }
+        String restPath = service.getPropertyString("uri");
+        ThingUID thingUID = getThingUID(service);
+        if (thingUID != null && url != null && restPath != null) {
+            String label = "openHAB server IP " + service.getHostAddresses()[0];
```

Comment:
Don't add the IP to the label, but make it the representation property instead.
```suggestion
            String label = "openHAB Server";
```

Reply:
If I do that, we will have no way to distinguish servers in the inbox.

Reply:
Here is how it looks now:
![image](https://user-images.githubusercontent.com/10186704/97095520-bbe5c080-1660-11eb-962c-353e18eaf30d.png)
With your proposed change, all entries would have the same name "openHAB server" in the inbox.

Reply:
Yes, that's exactly the idea - the label should describe the "what" it found and the presentation property allows to differentiate identical devices from each other.
@ghys, it seems then that the presentation property is not yet considered that way in the Inbox. Shall I enter an issue for that?

Reply:
@kaikreuzer : the representation property was not yet defined. With it added, I see the IP (representation property) on the second line.
@ghys : don't loose your time ;)

Reply:
Done (simplified label + representation property added)